visualization/app/codeCharta/ui/blacklistPanel/blacklistPanel.component.ts

```diff
@@ -19,8 +19,20 @@ export class BlacklistPanelController implements BlacklistSubscriber {
 	}
 
 	onBlacklistChanged(blacklist: BlacklistItem[]) {
-		this._viewModel.flatten = blacklist.filter(x => x.type === BlacklistType.flatten)
-		this._viewModel.exclude = blacklist.filter(x => x.type === BlacklistType.exclude)
+		// Use a set to uniquify the entries
```

Comment:
TODO: Check if this is indeed necessary or if it's not possible to add the same entry multiple times

Reply:
Good catch! While working on the code I somehow saw a string being added.

Reply:
We check for duplicates during insertion. That should be sufficient. Thus, I just use a regular array.